internal/cmd/stack/stack_selector.go

```diff
@@ -33,11 +36,25 @@ func getStack(cliCtx *cli.Context) (*stack, error) {
 		stackID := cliCtx.String(flagStackID.Name)
 		stack, err := stackGetByID(cliCtx.Context, stackID)
 		if err != nil {
+			if errors.Is(err, errNoStackFound) {
+				return nil, fmt.Errorf("stack with id %q could not be found. Please check that the stack exists and that you have access to it. To list available stacks run: spacectl stack list", stackID)
+			}
+
 			return nil, fmt.Errorf("failed to check if stack exists: %w", err)
 		}
-		if errors.Is(err, errNoStackFound) {
-			return nil, fmt.Errorf("stack with id %q could not be found. Please check that the stack exists and that you have access to it. To list available stacks run: spacectl stack list", stackID)
+
+		return stack, nil
+	} else if cliCtx.IsSet(flagRun.Name) {
+		runID := cliCtx.String(flagRun.Name)
+		stack, found, err := stackGetByRunID(cliCtx.Context, runID)
```

Comment:
why the double logic? just return the same error as we do in the previous method maybe?

Reply:
41ffe56627bc02752102acb40d82a3305c29e7f7